src/components/DatePicker/DatePicker.tsx

```diff
@@ -52,235 +48,217 @@ export interface BaseProps {
-        )
-      : intl.translate(`Polaris.DatePicker.months.${monthName(showNextMonth)}`);
-    const nextYear = multiMonth ? showNextToNextYear : showNextYear;
+  const setFocusDateAndHandleMonthChange = useCallback(
+    (date: Date) => {
+      if (onMonthChange) {
+        onMonthChange(date.getMonth(), date.getFullYear());
+      }
+      setHoverDate(date);
+      setFocusDate(date);
+    },
+    [onMonthChange],
+  );
 
-    const secondDatePicker = multiMonth ? (
-      <Month
-        onFocus={this.handleFocus}
-        focusedDate={focusDate}
-        month={showNextMonth}
-        year={showNextYear}
-        selected={deriveRange(selected)}
-        hoverDate={hoverDate}
-        onChange={this.handleDateSelection}
-        onHover={this.handleHover}
-        disableDatesBefore={disableDatesBefore}
-        disableDatesAfter={disableDatesAfter}
-        allowRange={allowRange}
-        weekStartsOn={weekStartsOn}
-      />
-    ) : null;
+  const handleDateSelection = useCallback(
+    (range: Range) => {
+      const {end} = range;
 
-    return (
-      <div
-        id={id}
-        className={styles.DatePicker}
-        onKeyDown={handleKeyDown}
-        onKeyUp={this.handleKeyUp}
-      >
-        <div className={styles.Header}>
-          <Button
-            plain
-            icon={ArrowLeftMinor}
-            accessibilityLabel={intl.translate(
-              'Polaris.DatePicker.previousMonth',
-              {
-                previousMonthName,
-                showPreviousYear,
-              },
-            )}
-            onClick={this.handleMonthChangeClick.bind(
-              null,
-              showPreviousMonth,
-              showPreviousYear,
-            )}
-          />
-          <Button
-            plain
-            icon={ArrowRightMinor}
-            accessibilityLabel={intl.translate('Polaris.DatePicker.nextMonth', {
-              nextMonth,
-              nextYear,
-            })}
-            onClick={this.handleMonthChangeClick.bind(
-              null,
-              showNextMonth,
-              showNextYear,
-            )}
-          />
-        </div>
-        <div className={styles.MonthContainer}>
-          <Month
-            onFocus={this.handleFocus}
-            focusedDate={focusDate}
-            month={month}
-            year={year}
-            selected={deriveRange(selected)}
-            hoverDate={hoverDate}
-            onChange={this.handleDateSelection}
-            onHover={this.handleHover}
-            disableDatesBefore={disableDatesBefore}
-            disableDatesAfter={disableDatesAfter}
-            allowRange={allowRange}
-            weekStartsOn={weekStartsOn}
-          />
-          {secondDatePicker}
-        </div>
-      </div>
-    );
-  }
-
-  private handleFocus = (date: Date) => {
-    this.setState({focusDate: date});
-  };
+      setHoverDate(end);
+      setFocusDate(new Date(end));
+      onChange(range);
```

Comment:
It looks like `onChange` was previously called when the setState for hover/focus had resolved. Any idea why, and do you think it'll change anything?

Reply:
Not sure, could just be trying to prevent race conditions. I expect this to be basically equivalent.

Do you think it'd be better to call onChange in a different `useEffect` function?

Reply:
I think we'll be ok, I was just wondering if anything stood out while you were working on this.